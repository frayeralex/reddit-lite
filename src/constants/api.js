export const URL = 'https://www.reddit.com';
export const SUB_REDDIT_PATH = '/r';
export const DEFAULT_SUB_REDDIT = 'reactjs';
export const DEFAULT_POST_LIMIT = 25;
