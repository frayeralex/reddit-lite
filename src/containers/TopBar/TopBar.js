import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { subreddit } from '../../store';
import './TopBar.scss';

export class TopBar extends React.PureComponent {
  static propTypes = {
    setCurrent: PropTypes.func,
    title: PropTypes.string,
  };

  static defaultProps = {
    setCurrent: () => {},
  };

  constructor(props) {
    super(props);

    this.state = {
      current: props.title,
    };
  }

  handleFormSubmit = event => {
    event.preventDefault();

    this.props.setCurrent(this.state.current);
  };

  handleNamedInputChange = ({ target: { value, name } }) => {
    this.setState({ [name]: value });
  };

  render() {
    return (
      <div className="TopBar">
        <div className="container">
          <div className="title">{this.props.title}</div>
          <form className="search-container" onSubmit={this.handleFormSubmit}>
            <input
              type="text"
              name="current"
              value={this.state.current}
              onChange={this.handleNamedInputChange}
            />
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  title: subreddit.selectors.getCurrent(state),
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setCurrent: subreddit.actions.setCurrent,
    },
    dispatch,
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TopBar);
