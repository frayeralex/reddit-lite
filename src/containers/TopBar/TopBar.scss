.TopBar {
  background-color: #42adf0;

  .container {
    margin: 0 auto;
    max-width: 1200px;
    height: 64px;
    padding: 18px 6px;
    display: flex;
  }

  .title {
    font-size: 22px;
    font-weight: 500;
    line-height: 26px;
    color: #fcfcfb;
  }
}
