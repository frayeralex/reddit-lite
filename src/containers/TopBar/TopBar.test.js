import React from 'react';
import { TopBar } from './TopBar';
import { DEFAULT_POST_LIMIT } from 'constants/api';

describe('<TopBar/>', () => {
  it('should render without crashing', function() {
    const component = shallow(<TopBar />);
    expect(component).toMatchSnapshot();
  });
  it('should correct root node', function() {
    const component = shallow(<TopBar />);
    expect(component.props().className).toEqual('TopBar');
  });
  it('should show title correct', function() {
    const title = 'title';
    const component = shallow(<TopBar />);
    expect(component.find('.title').text()).toEqual('');
    component.setProps({ title });
    expect(component.find('.title').text()).toEqual(title);
  });
  it('should handle input state changes correct', function() {
    const component = shallow(<TopBar />);
    component.setState({ current: 'current' });
    expect(component).toMatchSnapshot();
    component.setState({ current: '' });
    expect(component).toMatchSnapshot();
  });
  it('should handle form submit correct', function() {
    const setCurrent = jest.fn();
    const preventDefault = jest.fn();
    const component = shallow(<TopBar setCurrent={setCurrent} />);
    component.setState({ current: 'current' });
    const form = component.find('form').first();
    form.simulate('submit', { preventDefault });
    expect(preventDefault).toHaveBeenCalled();
    expect(setCurrent).toHaveBeenCalledWith('current');
  });
  it('should hide Prev button', function() {
    const props = {
      beforePostId: 'beforePostId',
      afterPostId: 'afterPostId',
    };
    const component = shallow(<TopBar {...props} />);
    expect(component).toMatchSnapshot();
    component.setProps({ beforePostId: null });
    expect(component).toMatchSnapshot();
    component.setProps({ afterPostId: null });
    expect(component).toMatchSnapshot();
  });
  it('should handle Prev Next button clicks', function() {
    const props = {
      beforePostId: 'beforePostId',
      afterPostId: 'afterPostId',
      fetchNextPage: jest.fn(),
      fetchPrevPage: jest.fn(),
    };
    const component = shallow(<TopBar {...props} />);
    expect(props.fetchPrevPage).not.toHaveBeenCalled();
    component
      .find('button')
      .at(0)
      .simulate('click');
    expect(props.fetchPrevPage).toHaveBeenCalled();
    expect(props.fetchNextPage).not.toHaveBeenCalled();
    component
      .find('button')
      .at(1)
      .simulate('click');
    expect(props.fetchNextPage).toHaveBeenCalled();
  });
});
