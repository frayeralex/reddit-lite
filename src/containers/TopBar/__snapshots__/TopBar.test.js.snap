// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar/> should handle input state changes correct 1`] = `
<div
  className="TopBar"
>
  <div
    className="container"
  >
    <div
      className="title"
    />
    <div
      className="controls-container"
    >
      <form
        className="search-container"
        onSubmit={[Function]}
      >
        <input
          name="current"
          onChange={[Function]}
          type="text"
          value="current"
        />
      </form>
      <div
        className="navigation-container"
      >
        <button
          onClick={[Function]}
        >
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<TopBar/> should handle input state changes correct 2`] = `
<div
  className="TopBar"
>
  <div
    className="container"
  >
    <div
      className="title"
    />
    <div
      className="controls-container"
    >
      <form
        className="search-container"
        onSubmit={[Function]}
      >
        <input
          name="current"
          onChange={[Function]}
          type="text"
          value=""
        />
      </form>
      <div
        className="navigation-container"
      >
        <button
          onClick={[Function]}
        >
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<TopBar/> should render without crashing 1`] = `
<div
  className="TopBar"
>
  <div
    className="container"
  >
    <div
      className="title"
    />
    <div
      className="controls-container"
    >
      <form
        className="search-container"
        onSubmit={[Function]}
      >
        <input
          name="current"
          onChange={[Function]}
          type="text"
        />
      </form>
      <div
        className="navigation-container"
      >
        <button
          onClick={[Function]}
        >
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;
